internal/handler/Handler.go

```
@@ -0,0 +1,276 @@
+		if err != nil {
+			http.Error(w, "Error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(p)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployee(w http.ResponseWriter, r *http.Request) {
+	matches := employeeRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	e, err := h.repo.GetEmployee(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("user not found"))
+		if err != nil {
+			http.Error(w, "Error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(e)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) createPosition(w http.ResponseWriter, r *http.Request) {
+	var p position.Position
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	h.repo.CreatePosition(p)
+	jsonBytes, err := json.Marshal(p)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) createEmployee(w http.ResponseWriter, r *http.Request) {
+	var e employee2.Employee
+	if err := json.NewDecoder(r.Body).Decode(&e); err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	h.repo.CreateEmployee(e)
```

If you try to add the same employee that code should return an error